services/frontend-service/src/ui/ReleaseDialog.tsx

```diff
@@ -265,18 +267,18 @@ const ReleaseEnvironment = (props: {
 }) => {
     const { overview, applicationName, version, environmentName } = props;
     // deploy
-    const [deployEnv, deployEnvState] = useUnaryCallback(
-        React.useCallback(
-            (api) =>
-                api.deployService().Deploy({
-                    application: applicationName,
-                    version: version,
-                    environment: environmentName,
-                    lockBehavior: LockBehavior.Ignore,
-                }),
-            [applicationName, version, environmentName]
-        )
-    );
+    const [deployEnv, deployEnvState] = useBatch({
```

Comment:
This can become tricky because you use `act` as dependency later. React uses object identity when comparing the dependencies and here you are creating a new object on every render.
I think the solution is to wrap the object in a useMemo like this:
```ts
const act = useMemo(() => +{
        action: {
            $case: 'deploy',
            deploy: {
                application: applicationName,
                version: version,
                environment: environmentName,
                ignoreAllLocks: false,
                lockBehavior: LockBehavior.Ignore,
            },
        },
    }, [applicationName, version, environmentName]);
const [deployEnv, deployEnvState] = useBatch(act);
```
Maybe Viktor can also help you.